### PR TITLE
Added invokables support to ParameterDeriverTrait

### PR DESCRIPTION
**Pull request by samdark**

### CODE REVIEW

File: src/ParameterDeriverTrait.php [hunk 1]
```diff
@@ -37,6 +37,9 @@ protected function getParameterType($callable) : string
                     $reflect = new \ReflectionObject($callable[0]);
                     $params = $reflect->getMethod($callable[1])->getParameters();
                     break;
+                case $this->isInvokeable($callable):
```
Comment (weierophinney) [inline src/ParameterDeriverTrait.php]: `s/isInvokeable/isInvokable/` here and everywhere else (proper spelling).

File: src/ParameterDeriverTrait.php [hunk 2]
```diff
@@ -111,4 +114,13 @@ protected function isClassCallable($callable) : bool
     {
         return (is_array($callable) && is_string($callable[0]) && class_exists($callable[0]));
     }
+
+    /**
+     * @param callable $callable
+     * @return True if the callable represents an invokeable object, false otherwise.
```
Comment (weierophinney) [inline src/ParameterDeriverTrait.php]: Honestly, these annotations aren't necessary, as they give no additional information beyond what the signature provides already. IDEs and phpdoc both pick up the typehints from the signature.

File: tests/Invokeable.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
+namespace Fig\EventDispatcher;
+
+class Invokeable
```
Comment (weierophinney) [inline tests/Invokeable.php]: Rename the file to `Invokable.php` and the class to `Invokable`.

**Review by weierophinney — CHANGES_REQUESTED** [review]
Main issue I've seen is the spelling of invokable (incorrectly spelled "invokeable" throughout the patch; see [wiktionary](https://en.wiktionary.org/wiki/invokable)).

Otherwise, functionality/behavior looks correct.

File: src/ParameterDeriverTrait.php (hunk 2 above)
Comment (samdark) [reply src/ParameterDeriverTrait.php]: That is in line with how other similar methods are documented.

Comment (weierophinney) [reply src/ParameterDeriverTrait.php]: If nothing else, fix the spelling: `s/invokeable/invokable/`. I'll defer to @Crell on whether or not to keep the docblock.

**Review by weierophinney — CHANGES_REQUESTED** [review]

**Review by weierophinney — APPROVED** [review]

File: src/ParameterDeriverTrait.php (hunk 2 above)
Comment (Crell) [reply src/ParameterDeriverTrait.php]: For consistency the docblock should remain, and have a description line.  Add that and I can merge.

**Pull request merged**